docs/src/description/declarations.md

```diff
@@ -1,24 +1,40 @@
 # Type declaration sections
 
-## Trace columns (`trace_columns`)
+## Constant (`const`)
+
+A `const` section contains a constant declaration. Constants can either be scalars, vectors or matrices. Each constant is a separate source section.
+
+**A `const` section is optional** and is conventionally declared at the top of the file however this is not enforced.
+
+Each constant is defined by an identifier and a value in the following format:
+
+```
+const FOO = 123
+const BAR = [1, 2, 3]
+const BAZ = [[1, 2, 3], [4, 5, 6]]
+```
+
+In the above example, FOO is a constant of type scalar with value `123`, BAR is a constant of type vector with value `[1, 2, 3]`, and BAZ is a constant of type matrix with value `[[1, 2, 3], [4, 5, 6]]`. The constant name should only contain uppercase letters otherwise an error is thrown.
+
+## Execution Trace (`trace_columns`)
 
 A `trace_columns` section contains declarations for `main` trace columns or `aux` (auxiliary) trace columns.
 
-The `main` and `aux` declarations define the shape of the main and auxiliary execution traces respectively and declare identifiers which can be used to refer to each of the columns in that trace.
+The `main` and `aux` declarations define the shape of the main and auxiliary execution traces respectively and define identifiers which can be used to refer to each of the columns or a group of columns in that trace. The columns can also be referred using built-in variables `\$main` and `\$aux` and the index of the column in the respective trace.
 
 **A `trace_columns` section with a `main` declaration is required for an AIR defined in AirScript to be valid.** The `aux` declaration is optional, but if it is defined then it must follow the `main` declaration.
 
 The following is a valid `trace_columns` block:
 
 ```
 trace_columns:
-    main: [a, b, c]
-    aux: [d, e]
+    main: [a, b, c[3], d]
+    aux: [e, f]
 ```
 
-In the above example, the main execution trace for the AIR has 3 columns which can be referenced by `a`, `b`, and `c`. Internally, these identifiers will reference the trace columns with indices 0, 1, and 2 respectively. Similarly, the auxiliary execution trace has 2 columns which can be referenced by `d` and `e`.
+In the above example, the main execution trace for the AIR has 6 columns where `a`, `b` and `d` are individual columns and `c` refers to a group of 3 columns. Single columns can be referenced using their identifiers (`a`, `b` and `d`) and columns in group `c` can be referenced using the identifier `c` and the index of the column within the group `c` (`c[0]`, `c[1]` and `c[2]`). Similarly, the auxiliary execution trace has 2 columns which can be referenced by `e` and `f`.
```

Comment:
One point of clarification, when we say "the main execution trace for the AIR has 6 columns", shouldn't it be columns instead of column bindings here?

Reply:
Yes, my mistake - you're correct about 6 columns. I don't quite agree with the rephrase though.

I would do something like:

"In the above example, the main execution trace for the AIR has 6 columns with 4 column bindings, where the identifiers `a`, `b`, and `d` are each bound to a single column and `c` refers to a group of 3 columns. Single columns can be referenced..."